### ISSUE TITLE
There is no way to reopen the main window upon killing it

### ISSUE THREAD

**Issue by abenkovskii** [akrasuski1/akrOS#39]

**Comment by akrasuski1** [comment]
What would you suggest? Adding `Main menu` entry to program list? This seems a little like inception, but whatever works :D

**Comment by abenkovskii** [comment]
Is it possible to make the main menu unkillable?

**Comment by akrasuski1** [comment]
Yes, it should be. Since main menu is opened first, it has PID=0 (or 1, whatever). I can add a rule: never kill pid=0 processs.
EDIT: IF its name is "Main menu" too.

**Comment by abenkovskii** [comment]
From the zen of Python: "special cases are not special enough to break the rules". I wonder if this rule is about situations like this.
P.S. I still don't see any better solution to this problem.

**Comment by akrasuski1** [comment]
Also, even if I make it unkillable, user can move it to background, and kill process manager, leaving menu unavailable. Perhaps this case would just mean user is stupid.

**Comment by abenkovskii** [comment]
We can:
1. make it unhideable too (we will also have to make sure that the window manager doesn't hide it)
2. exclude it from managed processes entirely
3. add a way to bring the main window back

**Comment by akrasuski1** [comment]
1. I don't like this for the reason stated in parentheses. The window manager is long enough already, I'm not going to add more special cases for that.
2. That is a quick&dirty solution and I wouldn't like to apply it, unless no other way shows up. The user might want to change menu location and not have it always sitting in upper lefet corner. It's getting boring there.
3. This would:
   (a) probably require extra action group as "rescue key"
   (b) have special cases too. What if all windows are already taken by some processes? Should one of them go to background? Uhhh.

To sum up, there does not seem to be a good way out of this. I would stay with semi-ignoring the issue (prevent killing, ignore hiding). The user should be clever enough to notice main menu is an important part of the infrastructure.

**Comment by abenkovskii** [comment]
We could add an option "open main menu here" to every empty window. What do you think?

**Comment by akrasuski1** [comment]
Yes - that is perfect UI.
4 maj 2015 09:22 "abenkovskii" notifications@github.com napisał(a):

**Comment by akrasuski1** [comment]
Also, I think that the title screen should gain new action group (8) as
"close this window/widget". This will be needed to be able to close one of
the menus if you spawn many of them.
4 maj 2015 10:15 "Adam Krasuski" akrasuski1@gmail.com napisał(a):

**Comment by akrasuski1** [comment]
Now that killing main menu is not equivalent to killing the whole system,
we should change end condition - add "finished" field to os_data and quit
when it is set, not when all processes are gone.
4 maj 2015 11:15 "Adam Krasuski" akrasuski1@gmail.com napisał(a):

**Comment by akrasuski1** [comment]
Done. Now if empty window is focused, it shows "Press AG9 to open main menu here" in status bar. When you press it, the new main menu is spawned in the focused window. The main menu itself gained "Close this window" option, to allow you to close redundant main menus. Finally, "Quit akrOS" option is much faster now, as it no longer redraws all the windows for the last time, leaving `clearscreen.` being the only redraw needed after the choice.
